xivo_dao/resources/user/persistor.py

```diff
@@ -69,6 +69,9 @@ def search(self, parameters):
         return SearchResult(total, users)
 
     def search_collated(self, parameters):
+        parameters = self.user_search._populate_parameters(parameters)
+        self.user_search._validate_parameters(parameters)
+        self.user_search.config.column_for_sorting(parameters['order'])
```

Comment:
This violates encapsulation.

Other possible solutions:
1. extract the validation logic from outside the `SearchSystem` and use it in both places.
2. push the manual collation logic down to `SearchSystem` (this one has my preference)